pkg/cmd/pulumi/convert/convert.go

```diff
@@ -457,13 +482,153 @@ func runConvert(
 	return nil
 }
 
-func newPluginContext(cwd string) (*plugin.Context, error) {
-	sink := diag.DefaultSink(os.Stderr, os.Stderr, diag.FormatOptions{
-		Color: cmdutil.GetGlobalColorization(),
-	})
-	pluginCtx, err := plugin.NewContext(sink, sink, nil, nil, cwd, nil, true, nil)
+// getPackagesToGenerateSdks parses the pcl files back in to read the package
+// blocks and for sdk generation.
+func getPackagesToGenerateSdks(
+	sourceDirectory string,
+) (map[string]*schema.PackageDescriptor, hcl.Diagnostics, error) {
+	// TODO use the package block to get the packages and parameterize them and generate SDKs.
+	files, err := os.ReadDir(sourceDirectory)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not read sourceDirectory: %w", err)
```

Comment:
Maybe this is more helpful for debugging?
```suggestion
		return nil, nil, fmt.Errorf("could not read %s: %w", sourceDirectory, err)
```